### PR TITLE
SONAR-6881 Disable OPTIONS, HEAD and TRACE methods of web server

### PR DESCRIPTION
**Pull request by simonbrandhof**
SONAR-6881 Disable OPTIONS, HEAD and TRACE methods of web server

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/platform/SecurityServletFilter.java [hunk 1]
```diff
@@ -19,34 +19,44 @@
...
 /**
  * This servlet filter sets response headers that enable browser protection against several classes if Web attacks.
  * The list of headers is mirrored in environment.rb as a workaround to Rack swallowing the headers..
  */
 public class SecurityServletFilter implements Filter {
 
-  @Override
+  private static final Set<String> ALLOWED_HTTP_METHODS = ImmutableSet.of("DELETE", "GET", "POST", "PUT");
+
   public void init(FilterConfig filterConfig) throws ServletException {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/platform/SecurityServletFilter.java]: MAJOR Add the "@Override" annotation above this method signature [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by sns-seb** [comment]
minor comment + fix SonarTech warning, otherwise LGTM

**Comment by simonbrandhof** [comment]
@sns-seb no need to notify me about sonar-tech issues !

**Pull request closed without merging**